Userland/Applications/Calendar/CalendarWidget.cpp

```diff
@@ -0,0 +1,298 @@
+    auto save_action = widget->create_save_action(save_as_action);
+
+    auto& file_menu = parent_window->add_menu("&File"_short_string);
+    file_menu.add_action(open_settings_action);
+    file_menu.add_action(new_calendar_action);
+    file_menu.add_action(open_calendar_action);
+    file_menu.add_action(save_as_action);
+    file_menu.add_action(save_action);
+
+    TRY(file_menu.try_add_separator());
+
+    TRY(file_menu.try_add_action(GUI::CommonActions::make_quit_action([](auto&) {
+        GUI::Application::the()->quit();
+    })));
+
+    auto& event_menu = parent_window->add_menu("&Event"_short_string);
+    event_menu.add_action(add_event_action);
+
+    auto view_menu = TRY(parent_window->try_add_menu("&View"_short_string));
+    TRY(view_menu->try_add_action(*view_month_action));
+    TRY(view_menu->try_add_action(*view_year_action));
+
+    auto help_menu = TRY(parent_window->try_add_menu("&Help"_short_string));
+    TRY(help_menu->try_add_action(GUI::CommonActions::make_command_palette_action(parent_window)));
+    TRY(help_menu->try_add_action(GUI::CommonActions::make_about_action("Calendar", TRY(GUI::Icon::try_create_default_icon("app-calendar"sv)), parent_window)));
+
+    return widget;
+}
+
+void CalendarWidget::create_on_events_change()
+{
+    m_event_calendar->event_manager().on_events_change = [&]() {
+        m_event_calendar->repaint();
+        window()->set_modified(true);
+        update_window_title();
+    };
+}
+
+void CalendarWidget::load_file(FileSystemAccessClient::File file)
+{
+    auto result = m_event_calendar->event_manager().load_file(file);
+    if (result.is_error()) {
+        GUI::MessageBox::show_error(window(), DeprecatedString::formatted("Cannot load file: {}", result.error()));
+        return;
+    }
+
+    window()->set_modified(false);
+    update_window_title();
+}
+
+NonnullRefPtr<GUI::Action> CalendarWidget::create_save_action(GUI::Action& save_as_action)
+{
+    return GUI::CommonActions::make_save_action([&](auto&) {
+        if (current_filename().is_empty()) {
+            save_as_action.activate();
+            return;
+        }
+
+        auto response = FileSystemAccessClient::Client::the().request_file(window(), current_filename().to_deprecated_string(), Core::File::OpenMode::Write);
+        if (response.is_error())
+            return;
+
+        auto result = m_event_calendar->event_manager().save(response.value());
+        if (result.is_error()) {
+            GUI::MessageBox::show_error(window(), DeprecatedString::formatted("Cannot save file: {}", result.error()));
+            return;
+        }
+
+        window()->set_modified(false);
+        update_window_title();
+    });
+}
+
+NonnullRefPtr<GUI::Action> CalendarWidget::create_save_as_action()
+{
+    return GUI::CommonActions::make_save_as_action([&](auto&) {
+        auto response = FileSystemAccessClient::Client::the().save_file(window(), "calendar", "json");
```

Comment:
I think `.json` as a file extension is a bit too generic and doesn't allow you to uniquely associate the file extension with the Calendar app.